app/controllers/v1/tasks_controller.rb

```diff
@@ -0,0 +1,116 @@
+module V1
+  class TasksController < ApplicationController
+    swagger_controller :tasks, 'Tasks'
+
+    swagger_api :index do
+      summary 'List all tasks'
+      notes 'This lists all the active tasks'
+      param :path, :project_id, :string, :required, 'Project Id'
+      param :query, :page, :integer, :optional, 'page number of results, default 1'
+      param :query, :page_size, :integer, :optional, 'number of results per page, default 25'
+    end
+    def index
+      tasks, errors = ListTasks.new(index_params).call
+      if errors.any?
+        render json: { errors: errors }, status: 400
+      else
+        render json: tasks
+      end
+    end
+
+    swagger_api :show do
+      summary 'Fetch a single Task'
+      param :path, :project_id, :string, :required, 'Project Id'
+      param :path, :id, :string, :required, 'Task Id'
+    end
+    def show
+      task = Task.find_by id: params[:id]
+      if task.present?
+        render json: task
+      else
+        render json: { errors: ['Task not found'] }, status: 404
+      end
+    end
+
+    swagger_api :create do
+      summary 'Creates a new Task or Tasks'
+      param :path, :project_id, :string, :required, 'Project Id'
+      param :form, :name, :string, :optional, 'Task name'
+      param :form, :description, :string, :optional, 'Task description'
+      param :form, :state, :string, :optional, 'Task state'
+    end
+    def create
+      project = Project.find_by id: params[:project_id]
+      task = project.present? ? project.tasks.create(task_params) : nil
+      if project.present? && task.present?
```

Comment:
Since you gate the creation of the task on the project being present, you don't need to check it again here.
That being said, just checking for presence on the task isn't enough because the way you created it, if create fails you will still get a task, it will just have errors on it and it won't be persisted.